lib/rules/structure/canonical.js

```diff
@@ -0,0 +1,23 @@
+const self = {
+    name: 'structure.canonical'
+,   section: 'metadata'
+,   rule: 'canonical'
+};
+
+exports.check = function (sr, done) {
+
+    const checkCanonical = function () {
+        var $lnk = sr.$("head > link[rel=canonical]");
+        if (!$lnk.length || !$lnk.attr("href")) sr.error(self, "not-found");
+    };
+
+    // That canonical link is mandatory starting from Oct 1, 2017.
+    // See https://lists.w3.org/Archives/Public/spec-prod/2017JulSep/0005.html
+    sr.transition({
+        to:          new Date('2017-09-30')
+    ,   doMeanwhile: function () { return done(); }
```

Comment:
Better

```js
    ,   doMeanwhile: () => {}
```

, as `done()` is called anyway before the end of function `check`.

Not sure if calling `done()` twice will distort the counter of &ldquo;pending checks&rdquo; and thus the progress bar on the UI. But just in case, and for simplicity.

Reply:
I updated the code with your suggestion! Thanks.